kura/test/org.eclipse.kura.core.net.test/src/test/java/org/eclipse/kura/core/net/util/NetworkUtilTest.java

```diff
@@ -0,0 +1,799 @@
+        assertEquals(1, result);
+    }
+
+    @Test
+    public void testGetNetmaskShortFormNullSubnet() {
+        try {
+            NetworkUtil.getNetmaskShortForm(null);
+            fail("exception was expected");
+        } catch (IllegalArgumentException e) {
+            // Success
+        }
+    }
+
+    @Test
+    public void testGetNetmaskShortFormEmptySubnet() {
+        try {
+            NetworkUtil.getNetmaskShortForm("");
+            fail("exception was expected");
+        } catch (IllegalArgumentException e) {
+            // Success
+        }
+    }
+
+    @Test
+    public void testGetNetmaskShortFormTooShortSubnet() {
+        try {
+            NetworkUtil.getNetmaskShortForm("255.255.255");
+            fail("exception was expected");
+        } catch (IllegalArgumentException e) {
+            // Success
+        }
+    }
+
+    @Test
+    public void testGetNetmaskShortFormTooLongSubnet() {
+        try {
+            NetworkUtil.getNetmaskShortForm("255.255.255.0.0");
+            fail("exception was expected");
+        } catch (IllegalArgumentException e) {
+            // Success
+        }
+    }
+
+    @Test
+    public void testGetNetmaskShortFormInvalidSubnet1() {
+        try {
+            NetworkUtil.getNetmaskShortForm("256.255.255.0");
+            fail("exception was expected");
+        } catch (IllegalArgumentException e) {
+            // Success
+        }
+    }
+
+    @Test
+    public void testGetNetmaskShortFormInvalidSubnet2() {
+        try {
+            NetworkUtil.getNetmaskShortForm("255.255.127.0");
+            fail("exception was expected");
+        } catch (IllegalArgumentException e) {
+            // Success
+        }
+    }
+
+    @Test
+    public void testDottedQuad() {
+        String result = NetworkUtil.dottedQuad(0xFFFFFFFF);
+        assertEquals("255.255.255.255", result);
+
+        result = NetworkUtil.dottedQuad(0xFFFFFF00);
+        assertEquals("255.255.255.0", result);
+
+        result = NetworkUtil.dottedQuad(0xFF120000);
+        assertEquals("255.18.0.0", result);
+
+        result = NetworkUtil.dottedQuad(0x80000000);
+        assertEquals("128.0.0.0", result);
+
+        result = NetworkUtil.dottedQuad(0x00000000);
+        assertEquals("0.0.0.0", result);
+    }
+
+    @Test
+    public void testConvertIp4Address() throws IllegalArgumentException {
+        int result = NetworkUtil.convertIp4Address("255.255.255.255");
+        assertEquals(0xFFFFFFFF, result);
+
+        result = NetworkUtil.convertIp4Address("255.255.255.0");
+        assertEquals(0xFFFFFF00, result);
+
+        result = NetworkUtil.convertIp4Address("255.18.0.0");
+        assertEquals(0xFF120000, result);
+
+        result = NetworkUtil.convertIp4Address("128.0.0.0");
+        assertEquals(0x80000000, result);
+
+        result = NetworkUtil.convertIp4Address("0.0.0.0");
+        assertEquals(0x00000000, result);
+    }
+
+    @Test
```

Comment:
I still think it would be much clearer if it would be:

```java
@Test(expected=IllegalArgumentException.class)
public void testConvertIp4AddressNullAddress() {
     NetworkUtil.convertIp4Address(null);
}
```